In [31]:


import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)


import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

import kagglehub


/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_customers_dataset.csv
/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_sellers_dataset.csv
/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_order_reviews_dataset.csv
/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_order_items_dataset.csv
/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_products_dataset.csv
/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_geolocation_dataset.csv
/kaggle/input/datasets/olistbr/brazilian-ecommerce/product_category_name_translation.csv
/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_orders_dataset.csv
/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_order_payments_dataset.csv


In [32]:

df_customers = pd.read_csv("/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_customers_dataset.csv")
df_sellers = pd.read_csv("/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_sellers_dataset.csv")
df_reviews = pd.read_csv("/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_order_reviews_dataset.csv")
df_items = pd.read_csv("/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_order_items_dataset.csv")
df_products = pd.read_csv("/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_products_dataset.csv")
df_geolocation = pd.read_csv("/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_geolocation_dataset.csv")
df_category = pd.read_csv("/kaggle/input/datasets/olistbr/brazilian-ecommerce/product_category_name_translation.csv")
df_orders = pd.read_csv("/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_orders_dataset.csv")
df_payments = pd.read_csv("/kaggle/input/datasets/olistbr/brazilian-ecommerce/olist_order_payments_dataset.csv")

In [33]:
datasets = {
    "Customers": df_customers,
    "Sellers": df_sellers,
    "Reviews": df_reviews,
    "Order Items": df_items,
    "Products": df_products,
    "Geolocation": df_geolocation,
    "Category Translation": df_category,
    "Orders": df_orders,
    "Payments": df_payments
}

for name, df in datasets.items():
    print(f"{name}: {df.shape}")

Customers: (99441, 5)
Sellers: (3095, 4)
Reviews: (99224, 7)
Order Items: (112650, 7)
Products: (32951, 9)
Geolocation: (1000163, 5)
Category Translation: (71, 2)
Orders: (99441, 8)
Payments: (103886, 5)


# Data Understanding & Data Quality Assessment

## Customers Table

In [34]:
df_customers.head()

,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP
1,18955e83d337fd6b2def6b18a428ac77,290c77bc529b7ac935b93aa66c333dc3,9790,sao bernardo do campo,SP
2,4e7b3e00288586ebd08712fdd0374a03,060e732b5b29e8181a18229c7b0b2b5e,1151,sao paulo,SP
3,b2b6027bc5c5109e529d4dc6358b12c3,259dac757896d24d7702b9acbbff3f3c,8775,mogi das cruzes,SP
4,4f2d8ab171c80ec8364f7c12e35b23ad,345ecd01c38d18a9036ed96c73b8d066,13056,campinas,SP


In [35]:
df_customers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


In [36]:
df_customers.isnull().sum()

customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

In [37]:
df_customers.nunique()

customer_id                 99441
customer_unique_id          96096
customer_zip_code_prefix    14994
customer_city                4119
customer_state                 27
dtype: int64

In [38]:
df_customers.duplicated().sum() 

np.int64(0)

In [39]:
df_customers['customer_id'].duplicated().sum() 

np.int64(0)

In [40]:
df_customers['customer_unique_id'].duplicated().sum() 

np.int64(3345)

In [41]:
df_customers['customer_id'].duplicated().sum() 

np.int64(0)

In [42]:
df_customers['customer_state'].value_counts() 

customer_state
SP    41746
RJ    12852
MG    11635
RS     5466
PR     5045
SC     3637
BA     3380
DF     2140
ES     2033
GO     2020
PE     1652
CE     1336
PA      975
MT      907
MA      747
MS      715
PB      536
PI      495
RN      485
AL      413
SE      350
TO      280
RO      253
AM      148
AC       81
AP       68
RR       46
Name: count, dtype: int64

In [43]:
df_customers['customer_state'].unique()

array(['SP', 'SC', 'MG', 'PR', 'RJ', 'RS', 'PA', 'GO', 'ES', 'BA', 'MA',
       'MS', 'CE', 'DF', 'RN', 'PE', 'MT', 'AM', 'AP', 'AL', 'RO', 'PB',
       'TO', 'PI', 'AC', 'SE', 'RR'], dtype=object)

In [44]:
df_customers['customer_city'].value_counts().head(20) 

customer_city
sao paulo                15540
rio de janeiro            6882
belo horizonte            2773
brasilia                  2131
curitiba                  1521
campinas                  1444
porto alegre              1379
salvador                  1245
guarulhos                 1189
sao bernardo do campo      938
niteroi                    849
santo andre                797
osasco                     746
santos                     713
goiania                    692
sao jose dos campos        691
fortaleza                  654
sorocaba                   633
recife                     613
florianopolis              570
Name: count, dtype: int64

In [45]:
df_customers['customer_city'].unique()[:50]

array(['franca', 'sao bernardo do campo', 'sao paulo', 'mogi das cruzes',
       'campinas', 'jaragua do sul', 'timoteo', 'curitiba',
       'belo horizonte', 'montes claros', 'rio de janeiro',
       'lencois paulista', 'caxias do sul', 'piracicaba', 'guarulhos',
       'pacaja', 'florianopolis', 'aparecida de goiania', 'santo andre',
       'goiania', 'cachoeiro de itapemirim', 'sao jose dos campos',
       'sao roque', 'camacari', 'resende', 'sumare', 'novo hamburgo',
       'sao luis', 'sao jose', 'santa barbara', 'ribeirao preto',
       'ituiutaba', 'taquarituba', 'sao jose dos pinhais', 'barrinha',
       'parati', 'dourados', 'trindade', 'cascavel', 'fortaleza',
       'brasilia', 'pelotas', 'porto alegre', 'salto', 'jundiai',
       'cacapava', 'sao vicente', 'uberlandia', 'botelhos', 'sao goncalo'],
      dtype=object)

**Leading Space Check**

In [46]:
(df_customers['customer_city'] != df_customers['customer_city'].str.strip()).sum()

np.int64(0)

**Mixed-case value check**

In [47]:
(df_customers['customer_city'] != df_customers['customer_city'].str.lower()).sum()

np.int64(0)

**Empty Strings Check**

In [48]:
(df_customers['customer_city'].str.strip() == '').sum()

np.int64(0)

### Insight_Customers Table :

- Total records: 99,441
- Total columns: 5
- Missing values: None
- Duplicate rows: None
- Duplicate customer_id values: None
- Duplicate customer_unique_id values: 3,345
- Customer state values: 27 unique states
- City formatting issues: None

**Conclusion:**

The customers table contains no missing values or completely duplicate
records. The customer_id column is unique.

Duplicate customer_unique_id values were retained because they can
represent multiple records associated with the same customer.

No formatting issues were found in the customer city or state columns.
Therefore, no cleaning was required for the customers table.

## Order Table